DualView/src/main/java/com/microsoft/device/display/samples/dualview/fragments/RestaurantsFragment.kt

```diff
@@ -0,0 +1,136 @@
+        if (currentScreenInfo?.isDualMode() == false) {
+            menu.clear()
+            inflater.inflate(R.menu.menu_list, menu)
+        }
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_map -> {
+                markSelectedRestaurant(NO_ITEM_SELECTED)
+                shopMapFragment(R.id.first_container_id)
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_restaurants, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupRestaurantList()
+    }
+
+    /**
+     * Setup for the restaurant [RecyclerView]
+     */
+    private fun setupRestaurantList() {
+        restaurantAdapter = RestaurantAdapter(requireContext(), selectedViewModel, ::onItemClick)
+        restaurants_recycler_view.adapter = restaurantAdapter
+        restaurants_recycler_view.setHasFixedSize(true)
+        val divider = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
+        ResourcesCompat.getDrawable(resources, R.drawable.item_divider, null)?.let {
+            divider.setDrawable(it)
+        }
+        restaurants_recycler_view.addItemDecoration(divider)
+    }
+
+    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
+        currentScreenInfo = screenInfo
+        requireActivity().invalidateOptionsMenu()
+        if (!screenInfo.isDualMode()) {
+            restaurantAdapter?.selectItem(NO_ITEM_SELECTED)
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        ScreenManagerProvider.getScreenManager().addScreenInfoListener(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
+    }
+
+    /**
+     * Called when a restaurant is selected
+     */
+    private fun onItemClick(item: Restaurant) {
+        val mapContainerId = when {
+            currentScreenInfo?.isDualMode() == true -> R.id.second_container_id
+            else -> R.id.first_container_id
+        }
+        markSelectedRestaurant(selectedViewModel.getItemPosition(item))
+        shopMapFragment(mapContainerId)
+    }
+
+    /**
+     * Adds some visual changes for the selected restaurant
+     */
+    private fun markSelectedRestaurant(position: Int) {
+        restaurantAdapter?.selectItem(position)
+    }
+
+    /**
+     * Adds the map fragment to the given container
+     *
+     * @param containerId The view id used as the container for the [MapFragment]
+     */
+    private fun shopMapFragment(@IdRes containerId: Int) {
```

Comment:
I think there's a small typo here, was it supposed to be "showMapFragment"?

Reply:
Done